pkg/run/run.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Entrypoint to run the app
+
+// Package run provides a function that can be invoked inside of main to set up
+// all the service-standard components we expect every service to run.
+//
+// clients should provide any runners they require as part of the app to the
+// Run function with OptAddRunner, and then call `Run` in their main function.
+package run
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/getoutreach/gobox/pkg/app"
+	"github.com/getoutreach/gobox/pkg/async"
+	"github.com/getoutreach/gobox/pkg/env"
+	"github.com/getoutreach/gobox/pkg/olog"
+	"github.com/getoutreach/gobox/pkg/trace"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/automemlimit"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/gomaxprocs"
+	"github.com/getoutreach/stencil-golang/pkg/serviceactivities/shutdown"
+
+	"github.com/getoutreach/httpx/pkg/handlers"
+)
+
+// runOpts are the options to pass to Run to configure your service
+type runOpts struct {
+	httpAppHandler http.Handler
+	log            *slog.Logger
+	httpAddr       string
+	runners        []async.Runner
+}
+
+// Option is the interface for an  option
+type Option interface {
+	apply(*runOpts) error
+}
+
+// optionFunc is a function that implements serviceOption
+type optionFunc func(*runOpts) error
+
+// apply implements serviceOption.
+func (s optionFunc) apply(o *runOpts) error {
+	return s(o)
+}
+
+// OptLogger sets the logger. Otherwise it defaults to olog.New
+func OptLogger(l *slog.Logger) Option {
```

Comment:
nit: the idiom I've see across the company and in the ecosystem for functions returning functional options are to prefix them with `With`. So `WithLogger`, `WithHTTPAppHandler`, etc

Reply:
I think that is fine too, I have a mild preference for Opt, although really I should make the interface an Opt too.

Reply:
Haha I don't really mind either way. Here are some hand wave-y metrics on usage here between the two if they make you think any differently - https://github.com/search?q=org%3Agetoutreach%20path%3A*.go%20%22func%20With%22&type=code
https://github.com/search?q=org%3Agetoutreach+path%3A*.go+%22func+Opt%22&type=code